compiler/src/iree/compiler/Utils/test/TileSizeSelectionTest.cpp

```diff
@@ -0,0 +1,51 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "iree/compiler/Codegen/Common/TileSizeSelection.h"
+#include "iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenDialect.h"
+
+using namespace mlir::iree_compiler;
+using namespace llvm;
+using namespace mlir;
+using namespace testing;
+
+TEST(TileSizeSelection, Basic) {
+  // 0. Init MLIR
+  // TODO: Move this to a test fixture
+  mlir::MLIRContext ctx;
+  DialectRegistry reg;
+  reg.insert<IREE::Codegen::IREECodegenDialect>();
+  ctx.appendDialectRegistry(reg);
+  ctx.loadAllAvailableDialects();
```

Comment:
It'd be very nice if we move it to SetUp()! We can do it in a follow-up

Putting a pointer as a note: https://github.com/google/googletest/blob/main/docs/primer.md#writing-the-main-function

Reply:
I just learned that we prefer using the constructor, this looks good to me!

https://github.com/google/googletest/blob/main/docs/faq.md#should-i-use-the-constructordestructor-of-the-test-fixture-or-setupteardown-ctorvssetup